controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -280,60 +177,24 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
-            @Override
-            public void connectionDropped() {
-                log.warn(requestId, "sealSegment {} connectionDropped", qualifiedName);
-                result.completeExceptionally(
-                        new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
-            }
-
-            @Override
-            public void wrongHost(WireCommands.WrongHost wrongHost) {
-                log.warn(requestId, "sealSegment {} wrongHost", qualifiedName);
-                result.completeExceptionally(
-                        new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
-            }
-
-            @Override
-            public void segmentSealed(WireCommands.SegmentSealed segmentSealed) {
-                log.info(requestId, "sealSegment {} segmentSealed", qualifiedName);
-                result.complete(true);
-            }
-
-            @Override
-            public void segmentIsSealed(WireCommands.SegmentIsSealed segmentIsSealed) {
-                log.info(requestId, "sealSegment {} SegmentIsSealed", qualifiedName);
-                result.complete(true);
-            }
-
-            @Override
-            public void processingFailure(Exception error) {
-                log.error(requestId, "sealSegment {} failed", qualifiedName, error);
-                handleError(error, result, type);
-            }
-
-            @Override
-            public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenCheckFailed) {
-                result.completeExceptionally(
-                        new WireCommandFailedException(new AuthenticationException(authTokenCheckFailed.toString()),
-                                type, WireCommandFailedException.Reason.AuthFailed));
-            }
-        };
-
-        WireCommands.SealSegment request = new WireCommands.SealSegment(requestId, qualifiedName, delegationToken);
-        sendRequestAsync(request, replyProcessor, result, ModelHelper.encode(uri));
-        return result;
+                                                   long clientRequestId) {
+        RawClient connection = new RawClient(ModelHelper.encode(uri), connectionFactory);
+        final long requestId = connection.getFlow().asLong();
+
+        Map<Class, SegmentHelperLog> expectedReplies = new HashMap<>();
+        expectedReplies.put(WireCommands.SegmentSealed.class,
+                () -> log.info(requestId, "sealSegment {} segmentSealed", qualifiedName));
+        expectedReplies.put(WireCommands.SegmentIsSealed.class,
+                () -> log.info(requestId, "sealSegment {} SegmentIsSealed", qualifiedName));
+
+        return connection.sendRequest(requestId, new WireCommands.SealSegment(requestId, qualifiedName, delegationToken))
+                .thenApply(r -> transformReply(r, expectedReplies, connection));
     }
 
     public CompletableFuture<UUID> createTransaction(final String scope,
```

Comment:
Any specific reason for excluding `transaction` related methods, `updatePolicy` from following the same pattern as rest of the methods?

Reply:
this modification is for the functions that return CompleatbleFuture<Boolean>, the left part will be commit after this get merged.